syntax = "proto3";

import "messages.proto";


service GraphFgaService {
    rpc store_view(StoreViewRequest) returns (StoreViewResponse) {}
    rpc store_create(StoreCreateRequest) returns (StoreCreateResponse) {}
    rpc store_update(StoreUpdateRequest) returns (StoreUpdateResponse) {}

    rpc store_read(StoreReadRequest) returns (StoreReadResponse) {}
    rpc store_write(StoreWriteRequest) returns (StoreWriteResponse) {}
    rpc store_check(StoreCheckRequest) returns (StoreCheckResponse) {}
    rpc store_list_objects(StoreListObjectsRequest) returns (StoreListObjectsResponse) {}
}